### PR TITLE
[2.13.x] DDF-4239 Upgrades checkstyle

### PR DESCRIPTION
**Pull request by coyotesqrl**
#### What does this PR do?
Upgrades checkstyle and ddf-support.

Depends on codice/ddf-support#56

#### Who is reviewing it? 

@paouelle @brjeter 
#### Select relevant component teams: 
N/A

#### Ask 2 committers to review/merge the PR and tag them here.

@clockard
@shaundmorris

#### How should this be tested?
A full CI build is sufficient. This is only a build-time validation tool.

#### Any background context you want to provide?
Later versions of Checkstyle are needed to support upgrading to Java 11.

#### What are the relevant tickets?
[DDF-4239](https://codice.atlassian.net/browse/DDF-4239)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
         <reports.repository.url />
         <project.scm.id>github</project.scm.id>
         <!-- Must MANUALLY update this if the ddf/support project's version changes -->
-        <ddf.support.version>2.3.14</ddf.support.version>
+        <ddf.support.version>2.3.16</ddf.support.version>
```
Comment (coyotesqrl) [inline pom.xml]: @paouelle, I see that this branch had been on version 2.3.14, not 2.3.15, of ddf-support. Are there any potential issues that might arise from 2.3.15 that need to be reflected here?

Comment (paouelle) [reply pom.xml]: If I look at what changes between 2.14 and 2.16, only the artifact enforcer was added. So you should be able to answer your own question 😄

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Pull request merged**